[docs] revamp database AWS IAM reference

GavinFrazar opened the pull request:
Context
- https://github.com/gravitational/teleport/issues/43829

This is a docs only PR that:
- replaces the dynamic IAM permissions setup with simple permissions
- moves the discovery/database service permissions into partials
- adds a reference section for the discovery service
- removes the IAM permissions for discovery from the database access IAM reference

I wanted to use partials so I can easily inline the permissions and explanations for those permissions in future updates that will replace (or perhaps just supplement) the typical bootstrap command step in our guides.

greedy52 commented on docs/pages/includes/discovery/reference/aws-iam/elasticache.mdx:
```diff
@@ -0,0 +1,28 @@
+| Statement | Purpose |
+| --------- | ------- |
+| `ElastiCacheDiscovery` (required) | Discover ElastiCache replication groups. |
+| `ElastiCacheFetchMetadata` (optional) | Import AWS tags and additional metadata for each database as Teleport database labels. |
```
I am not sure how I feel about marking these optional. Without importing tags, you must have a wildcard matcher. I recall helping a customer debug an issue where RDS proxies were not discovered and turned out to be missing permissions to fetch tags. Even not for tag filtering, I do not see why we wouldn't import more labels to provide better context. 

For simplicity, I would group all of these in one as required. For the API that describes extra endpoints, maybe we can have a paragraph (similar to the one in RDS section) that the API is only required when you want to discover those specific endpoints. 

Or maybe the word "optional" is too weak. 

Let me know what you think.

GavinFrazar replied:
I'm glad you pointed that out - I assumed that required means "it can't possibly work without these permissions", whereas optional it technically could work without those permissions, but you are right that if the db service relies on imported tags in its matcher then "optional" is misleading.
I will just remove the required/optional phrasing.

I still want to keep the separate statements though, because I want to describe the different purposes of each.

GavinFrazar commented on docs/pages/auto-discovery/reference/aws-iam.mdx:
```diff
@@ -0,0 +1,52 @@
+Service instance can use.
+
+Each section below describes the IAM permissions required to discover a specific
+type of AWS resource.
```
Included

GavinFrazar replied:
I also sorted the database types alphabetically and put them under a ## Databases heading, with the ToC depth set to 3:

![image](https://github.com/user-attachments/assets/cd5efc2f-5311-4c3e-aa62-f5b8117d0db2)

GavinFrazar commented on docs/pages/auto-discovery/reference/configuration.mdx:
```diff
@@ -0,0 +1,10 @@
+---
+title: Discovery Service Configuration Reference
+description: Configuration reference for the Teleport Discovery Service.
```
removed

GavinFrazar committed:
revamp database AWS IAM reference

* move the discovery/database service permissions into partials
* replace the dynamic IAM permissions setup with simple permissions
* add a reference section for the discovery service
* remove details wrappers from IAM role trust setup partial

GavinFrazar commented:
TIL: if you force push your branch to match master, github closes the PR automatically

ptgott reviewed (approved):
In `docs/pages/enroll-resources/database-access/reference/aws.mdx`, I think the approach of placing `Tabs` components inside `Details` boxes makes the page more difficult to read, since it requires a lot of user interaction to find the right instructions.

What would you think of removing the `Details` boxes and replacing them with an ordered list instead?

```markdown
1. Configure trust relationships on dynamodb-user-role

   <Tabs>
   </Tabs>

1. Configure permissions policies on teleport-db-service-role

   <Tabs>
   </Tabs>
```

GavinFrazar commented:
> In `docs/pages/enroll-resources/database-access/reference/aws.mdx`, I think the approach of placing `Tabs` components inside `Details` boxes makes the page more difficult to read, since it requires a lot of user interaction to find the right instructions.
> 
> What would you think of removing the `Details` boxes and replacing them with an ordered list instead?
> 
> ```
> 1. Configure trust relationships on dynamodb-user-role
> 
>    <Tabs>
>    </Tabs>
> 
> 1. Configure permissions policies on teleport-db-service-role
> 
>    <Tabs>
>    </Tabs>
> ```

I like that better, thanks for suggesting! I've removed the details in https://github.com/gravitational/teleport/compare/8131761c4dc63b292c3445bcc53f820acf9daee0..67766c31673678778471662541a1aeac154b99bd#diff-8bd1e9fd317807a8aa0ea368afb89b151e9ca37a1ba7dda68a72a9119c01371e

These role trust setup steps are all in a partial, and we included that partial in an AWS troubleshooting partial that was a little too crowded to keep the steps outside of details. For that specific case, I've wrapped the role trust partial in a single Details block: https://github.com/gravitational/teleport/compare/8131761c4dc63b292c3445bcc53f820acf9daee0..67766c31673678778471662541a1aeac154b99bd#diff-8bd1e9fd317807a8aa0ea368afb89b151e9ca37a1ba7dda68a72a9119c01371eR48-R52

I also sorted the AWS IAM reference in the database access page and fixed some redshift/redshift serverless info that I noticed.
